### PR TITLE
[stable/envoy] Fix deployment spec

### PR DESCRIPTION
**Pull request by hypnoglow**
#### What this PR does / why we need it:

- Remove serviceName as it is not defined for Deployment

- Move terminationGracePeriodSeconds from Deployment spec to pod
template spec

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

**Comment by hypnoglow** [comment]
/assign @foxish

Ref. maintainers from Chart.yaml: @josdotso

File: stable/envoy/templates/deployment.yaml [hunk 1]
```diff
@@ -44,6 +40,7 @@ spec:
 {{ toYaml .Values.securityContext | indent 8 }}
 {{- if .Values.priorityClassName }}
       priorityClassName: "{{ .Values.priorityClassName }}"
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
```
Comment (unguiculus) [inline stable/envoy/templates/deployment.yaml]: Should be outside the if statement.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/envoy/templates/deployment.yaml (hunk 1 above)
Comment (hypnoglow) [reply stable/envoy/templates/deployment.yaml]: woops.. fixed.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**